internal/service/cloudbackupschedule/resource_cloud_backup_schedule.go

```diff
@@ -343,8 +344,12 @@ func resourceRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Di
 	projectID := ids["project_id"]
 	clusterName := ids["cluster_name"]
 
-	backupPolicy, _, err := connV2.CloudBackupsApi.GetBackupSchedule(context.Background(), projectID, clusterName).Execute()
+	backupPolicy, resp, err := connV2.CloudBackupsApi.GetBackupSchedule(context.Background(), projectID, clusterName).Execute()
 	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound {
+			d.SetId("")
```

Comment:
why in here we don't do `resp.State.RemoveResource(ctx)` ?

Reply:
same question, what is it for and why we use it only in some resources

Reply:
This is the difference between SDK v2 and TPF, see [google doc](https://docs.google.com/document/d/1Q0ePD_hR5UzoyZR1tdP24cf438Il3UcmHmYawseGvkQ/edit#bookmark=id.k0ep5xjcyouv)